src/components/CalendarPicker/index.js

```diff
@@ -0,0 +1,162 @@
+import _ from 'underscore';
+import React from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import moment from 'moment';
+import Text from '../Text';
+import ArrowIcon from './ArrowIcon';
+import styles from '../../styles/styles';
+import {propTypes as calendarPickerPropType, defaultProps as defaultCalendarPickerPropType} from './calendarPickerPropTypes';
+import generateMonthMatrix from './generateMonthMatrix';
+import withLocalize from '../withLocalize';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import CONST from '../../CONST';
+
+class CalendarPicker extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.monthNames = moment.localeData(props.preferredLocale).months();
+        this.daysOfWeek = moment.localeData(props.preferredLocale).weekdays();
+
+        let currentDateView = props.value;
+        if (props.selectedYear) {
+            currentDateView = moment(currentDateView).set('year', props.selectedYear).toDate();
+        }
+        if (props.maxDate < currentDateView) {
+            currentDateView = props.maxDate;
+        } else if (props.minDate > currentDateView) {
+            currentDateView = props.minDate;
+        }
+
+        this.state = {
+            currentDateView,
+        };
+
+        this.moveToPrevMonth = this.moveToPrevMonth.bind(this);
+        this.moveToNextMonth = this.moveToNextMonth.bind(this);
+        this.onYearPickerPressed = this.onYearPickerPressed.bind(this);
+        this.onDayPressed = this.onDayPressed.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.minDate <= this.props.maxDate) {
+            return;
+        }
+        throw new Error('Minimum date cannot be greater than the maximum date.');
+    }
+
+    componentDidUpdate(prevProps) {
+        // Check if selectedYear has changed
+        if (this.props.selectedYear === prevProps.selectedYear) {
+            return;
+        }
+
+        // If only the selectedYear prop has changed, update the currentDateView state with the new year value
```

Comment:
I thinkkk having `only` here is incorrect :D

```suggestion
        // If the selectedYear prop has changed, update the currentDateView state with the new year value
```